doc/vlan/SONiC_vlan_translation_qnq.md

```diff
@@ -0,0 +1,1025 @@
+
+CVLANs on both the ends are same
+![VLAN translate use case with VxLAN](./images/Map_CVLAN_VNI.png)
+
+#### Usecase - 3 VLAN Translation with VxLAN
+   * Ingress - Map single/double C-TAGs to VNI - 3rd tag present in the packet is treated as payload
+   * Egress  - Map VNI to single/double C-TAGs
+
+CVLANs on both the ends are different
+![VLAN translate use case with VxLAN](./images/MAP_CVLAN_DTAG_VNI.png)
+
+
+### 1.1.2 Q-in-Q Tunneling usecases
+#### Usecase - 1 Native 
+
+![qinq tunnel usecase -1](./images/qinq_tunnel_usecase_1.png)
+
+#### Usecase -2 Q-in-Q Tunnel with VxLAN
+
+![qinq tunnel usecase -2](./images/qinq_tunnel_usecase_2.png)
+
+### 1.1.3 point-to-point usecases
+
+![Point-to-Point usecases -1](./images/Point_to_point_single_Switch_1.png)
+![Point-to-Point usecases -2](./images/Point_to_point_single_Switch.png)
+
+### 1.1.4 VxLAN DCI usecase
+![VXLAN DCI usecase](./images/vxlan_dci_usecase.png)
+
+## 1.2 Requirements
+### 1.2.1 Functional requirements for 4.1
+* Ability to translate a customer VLAN (single & double tagged) to Provider VLAN and forward based on Provider VLAN
+   * point-to-point use cases - the traffic ingresses from one port and egress on to a different port within the same PE.
+* Ability to translate a customer VLAN (single & double tagged) to VxLAN VNI and forward in the overlay network
+* Ability to tunnel Customer VLANs across a Provider Network
+   * point-to-point use cases - the traffic ingresses from one port and egress on to a different port within the same PE.
+* Ability to tunnel Customer VLANs across a Overlay (VxLAN) Network
+* Support forwarding L2 traffic with VLAN translation and Q-in-Q schemes
+* Support VLAN stacking & Translation schemes on both Ethernet & PortChannel Interfaces
+
+#### Future requirements
+* Ability to configure standard & custom EtherType(TPID) on the outer Tags
+* Hair-pin use case - Traffic ingresses with a VLAN Stack and egresses with a different VLAN stack on the same port. (swap outer vlan and keep the inner vlan intact)
+* Ability to route traffic on SVLAN incase of VLAN Translation (Layer 3 support)
+* L2 protocol packet(ex:STP, IGMP) passthrough in case of QinQ/VLAN stacking use cases
+* Ability to provide multi-homing support via MCLAG on custuomer facing interfaces
+* Supporting VLAN stacking & translation functionality on SONiC VS image
+
+### 1.2.2 Configuration requirements
+* Support confguration of VLAN mapping rules to implement VLAN Stacking(Q-in-Q) on both Ethernet & PortChannel Interfaces
+* Support confguration of VLAN mapping rules to implement VLAN translation on both Ethernet & PortChannel Interfaces
+* Provide various validations to implement both the VLAN mapping schemes
+* Provide configuraiton interfaces via REST, gNMI and KLISH.
+
+### 1.2.3 Hardware requirements
+* Support VLAN stacking & translation on TD3 & TD4 platforms
+
+### 1.2.4 Scaling requirements
+* Support 2k service chains (both VLAN stacking and translation schemes) per system
```

Comment:
A series of translation or Stacking scheme across the network (end-to-end) is considered a service chain.
Since this involves VXLAN as well, we restricted it to 2k.